BlueDBOnDisk/src/main/java/io/bluedb/disk/segment/Segment.java

```diff
@@ -175,6 +229,17 @@ private void moveRolledUpFileAndDeleteSourceFiles(Path newRolledupPath, Path tem
 		}
 	}
 
+	public static List<Range> getAllFileRangesInOrder(Path segmentPath) {
+		File segmentFolder = segmentPath.toFile();
+		List<File> allFilesInSegment = FileManager.getFolderContents(segmentFolder);
+		List<Range> allFileRanges = allFilesInSegment.stream()
+				.map( (File fl) -> Range.fromUnderscoreDelmimitedString(fl.getName()) )
+				.filter((r) -> r != null)
+				.collect(Collectors.toList());
+		Collections.sort(allFileRanges);
+		return allFileRanges;

```

Comment:
I'd do the stream like this (in case you're interested):
```
List<Range> allFileRanges = allFilesInSegment.stream()
				.map( (File fl) -> Range.fromUnderscoreDelmimitedString(fl.getName()) )
				.filter(Objects::nonNull)
				.sorted()
				.collect(Collectors.toList());
```

Reply:
Or better yet, you could make the Range.from method take a File object and could make it look like this:
```
List<Range> allFileRanges = allFilesInSegment.stream()
				.map(Range::fromFileWithUnderscoreDelmimitedName)
				.filter(Objects::nonNull)
				.sorted()
				.collect(Collectors.toList());
```

Reply:
sounds good, making changes.